apps/src/gamelab/GameLabP5.js

```diff
@@ -97,34 +97,37 @@ GameLabP5.prototype.init = function (options) {
   // positions before reporting touch coordinates
   //
   // NOTE: _updateNextTouchCoords() is nearly identical, but calls a modified
-  // getTouchInfo() function below that can return undefined
+  // getTouchInfo() function below that scales the touch postion with the play
+  // space and can return undefined
   window.p5.prototype._updateNextTouchCoords = function (e) {
-    if (e.type === 'mousedown' ||
-        e.type === 'mousemove' ||
+    var x = this.touchX;
+    var y = this.touchY;
+    if (e.type === 'mousedown' || e.type === 'mousemove' ||
         e.type === 'mouseup' || !e.touches) {
-      this._setProperty('_nextTouchX', this._nextMouseX);
-      this._setProperty('_nextTouchY', this._nextMouseY);
+      x = this.mouseX;
+      y = this.mouseY;
     } else {
       if (this._curElement !== null) {
         var touchInfo = getTouchInfo(this._curElement.elt, e, 0);
-
         if (touchInfo) {
-          this._setProperty('_nextTouchX', touchInfo.x);
-          this._setProperty('_nextTouchY', touchInfo.y);
+          x = touchInfo.x;
+          y = touchInfo.y;
         }
 
         var touches = [];
-        var touchIndex = 0;
         for (var i = 0; i < e.touches.length; i++) {
-          touchInfo = getTouchInfo(this._curElement.elt, e, i);
-          if (touchInfo) {
-            touches[touchIndex] = touchInfo;
-            touchIndex++;
-          }
+          touches[i] = getTouchInfo(this._curElement.elt, e, i);
```

Comment:
The previous code would eliminate touches from the array that are out of bounds and compact the array. This code will leave empty blanks in the middle of the `touches` array. Any reason for the change?